components/definition/cellRules/sections/mechanisms/NetPyNENewMechanism.js

```diff
@@ -1,38 +1,72 @@
 import React, { Component } from 'react';
-import IconButton from 'material-ui/IconButton';
+import Menu from 'material-ui/Menu';
+import Popover from 'material-ui/Popover';
+import MenuItem from 'material-ui/MenuItem';
 import ContentAdd from 'material-ui/svg-icons/content/add';
-import FontIcon from 'material-ui/FontIcon';
-
-const hoverColor = '#66d2e2';
-const changeColor = 'rgb(0, 188, 212)';
+import FloatingActionButton from 'material-ui/FloatingActionButton';
+import Utils from '../../../../../Utils';
 
 export default class NetPyNENewMechanism extends React.Component {
 
   constructor(props) {
     super(props);
     this.state = {
+      open: false,
+      mechanisms: []
     };
-
     this.handleClick = this.handleClick.bind(this);
+    this.handleButtonClick = this.handleButtonClick.bind(this);
+  };
+  
+  componentDidMount() {
+    var menuItems = []
+    Utils
+      .sendPythonMessage("netpyne_geppetto.getAvailableMechs", [])
+      .then((response) => {
+        response.forEach((item) => 
+          menuItems.push(<MenuItem key={item} value={item} primaryText={item}/>)
+        )
+        this.setState({mechanisms: menuItems})
+      })
+  };
+  
+  handleButtonClick = (event) => {
+    // This prevents ghost click.
+    event.preventDefault();
+    this.setState({
+      open: true,
+      anchorEl: event.currentTarget,
```

Comment:
This look to me a bit strange. If I am not wrong the anchorEl is gonna be the same one always right? Can you find another way to set this in the Popover component that is the only place where this variable is used. It is good to keep this.state with the minimum number of variables.

Can you please also update [this card in trello](https://trello.com/c/uOVcGmO8/201-revisit-plot-configuration-design) so we take care of the re-design (if needed) of this section as part of that card?

Reply:
The anchorEl takes a DOM component to render on top of it. So, we need to pass the DOM component created by the FloatingActionButton. I did not find a easier way to do it. I should work on it, but I'll do it in June. Right now, I don't have much time

Reply:
Now that I am reading the material-ui documentation I think you are right. I thought it was not particularly react-ish but it is how they do it. Sorry for that!